cmake/modules/FindQatDrv.cmake

```diff
@@ -0,0 +1,122 @@
+# - Find_QatDrv
+# Find the qat driver library and includes
+#
+# QatDrv_INCLUDE_DIRS = where to find cap.h, qae_mem.h, etc
+# QatDrv_LIBRARIES - List of libraries when using qat driver
+# QatDrv_FOUND - True if qat driver found
+
+set(expected_version "v1.7.L.4.14.0-00031")
+set(expected_version_url "https://www.intel.com/content/www/us/en/download/19081/intel-quickassist-technology-intel-qat-driver-for-linux-for-intel-server-boards-and-systems-based-on-intel-62x-chipset.html")
+
+function(find_qatdrv_source_dir)
+  if(NOT DEFINED ENV{ICP_ROOT})
+    set(expected_source_dir "/opt/APP/driver/QAT")
+    if(NOT EXISTS ${expected_source_dir})
+      meessage(FATAL_ERROR "${expected_source_dir} not found. "
+      	       "Please set an environment variable \"ICP_ROOT\" for the QAT driver package root directory, "
+	       "i.e. \"export ICP_ROOT=/the/directory/of/QAT\". "	
+	       "Or download and install QAT driver ${expected_version} manually at the link "
+	       "${expected_version_url}. "
+	       "Remember to set an environment variable for the package root directory, "
+               "i.e. \"export ICP_ROOT=/opt/APP/driver/QAT\"..")
+    else()
+      set(ENV{ICP_ROOT} ${expected_source_dir})
+    endif()
+  endif()
+endfunction()
+
+function(check_qatdrv_version expected_version expected_version_url)
+  string(REGEX REPLACE "^v1.7.L.([0-9]+.[0-9]+.[0-9]+).[0-9]+$"
+	 "\\1" EXPECTED_version "${expected_version}")
+  set(source_dir $ENV{ICP_ROOT})
+  set(versionfile "${source_dir}/versionfile")
+  if(NOT EXISTS ${versionfile})
+    message(FATAL_ERROR "${versionfile} not found. Please either \"rm -rf ${source_dir}\" "
+            "so you can download and install QAT driver ${expected_version} manually at the link "
+            "${expected_version_url}, or make sure ${source_dir} "
+            "contains a full copy of QAT driver ${expected_version}. "
```

Comment:
we have two copies of this error messages. might want to consolidate them after calling `find_package_handle_standard_args()`, and if `QatDrv_FOUND` is not `TRUE`.